cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -329,6 +333,11 @@ protected void rebalanceForBroker(Broker broker,
                                     ClusterModel clusterModel,
                                     Set<Goal> optimizedGoals,
                                     OptimizationOptions optimizationOptions) {
+
+    if (clusterModel.deadBrokers().isEmpty() && allAliveBrokersBalanced(clusterModel, optimizationOptions)) {
+      return;
+    }
+
```

Comment:
What if there are dead disks -- i.e. broker is alive, but there are dead disks on alive brokers?
Wouldn't this cause us to drop JBOD support when there are offline disks?

Reply:
Do you mean that if there is any alive broker having bad disks, we should continue with the normal rebalance process instead of returning here?

Reply:
Yes. If we terminate early due to this new if statement, after the goal optimization we would end up with offline replicas on dead disks, don't we?
But adding this if statement causes other issues. In particular, brokers excluded for replica move no longer give away their replicas to other available brokers. Let's discuss a potential approach to resolve this issues offline.